Running tanzu apps workload commands when KUBECONFIG is not set or pointing to a cluster that is not accessible/deleted returns a panic error stacktrace

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may delay the bug being addressed. Thanks!
-->

### Please fill out the issue checklist below and provide ALL the requested information.

- [x] I reviewed open and closed Github issues that may be related to my problem.
- [x] I am reporting a bug that others will be able to reproduce.

### Describe the bug
Running `tanzu apps workload` commands when KUBECONFIG is not set or pointing to a cluster that is not accessible/deleted returns a panic error stacktrace.


### Running `tanzu apps workload list` when my KUBECONFIG is not set/empty

```
panic: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable

goroutine 1 [running]:
github.com/vmware-tanzu/apps-cli-plugin/pkg/cli-runtime.(*client).lazyLoadDefaultNamespaceOrDie(0xc0005a8f80)
	/Users/adhol/projects/apps-cli-plugin/pkg/cli-runtime/client.go:215 +0x77
github.com/vmware-tanzu/apps-cli-plugin/pkg/cli-runtime.(*client).DefaultNamespace(0xc00013a010)
	/Users/adhol/projects/apps-cli-plugin/pkg/cli-runtime/client.go:52 +0x19
github.com/vmware-tanzu/apps-cli-plugin/pkg/cli-runtime.NamespaceFlag.func1(0xc000b0a000, {0x54363e8, 0x0, 0x0})
	/Users/adhol/projects/apps-cli-plugin/pkg/cli-runtime/flags.go:64 +0x5c
github.com/spf13/cobra.(*Command).execute(0xc000b0a000, {0x54363e8, 0x0, 0x0})
	/Users/adhol/projects/apps-cli-plugin/vendor/github.com/spf13/cobra/command.go:845 +0x5a8
github.com/spf13/cobra.(*Command).ExecuteC(0xc000216280)
	/Users/adhol/projects/apps-cli-plugin/vendor/github.com/spf13/cobra/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/adhol/projects/apps-cli-plugin/vendor/github.com/spf13/cobra/command.go:902
github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/command/plugin.(*Plugin).Execute(...)
	/Users/adhol/projects/apps-cli-plugin/vendor/github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/command/plugin/plugin.go:54
main.main()
	/Users/adhol/projects/apps-cli-plugin/cmd/plugin/apps/main.go:127 +0xa91
Error: exit status 2

✖  exit status 2
```


### Expected behavior
```
Error: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
Error: exit status 2

✖  exit status 2
```

### Running `tanzu apps workload list` when my KUBECONFIG is pointing to a cluster that is not accessible/deleted

```
panic: Get "https://127.0.0.1:50471/api?timeout=32s": dial tcp 127.0.0.1:50471: connect: connection refused

goroutine 1 [running]:
github.com/vmware-tanzu/apps-cli-plugin/pkg/cli-runtime.(*client).lazyLoadClientOrDie(0xc0005a9e80)
	/Users/adhol/projects/apps-cli-plugin/pkg/cli-runtime/client.go:203 +0x9b
github.com/vmware-tanzu/apps-cli-plugin/pkg/cli-runtime.(*client).Client(...)
	/Users/adhol/projects/apps-cli-plugin/pkg/cli-runtime/client.go:64
github.com/vmware-tanzu/apps-cli-plugin/pkg/cli-runtime.(*client).List(0xc0005a9e80, {0x3a951c8, 0xc000d2df80}, {0x3add610, 0xc000217730}, {0xc000fb44c0, 0x2, 0x2})
	/Users/adhol/projects/apps-cli-plugin/pkg/cli-runtime/client.go:77 +0x18c
github.com/vmware-tanzu/apps-cli-plugin/pkg/commands.(*WorkloadListOptions).Exec(0xc000cc2f90, {0x3a951c8, 0xc000d2df80}, 0xc000614320)
	/Users/adhol/projects/apps-cli-plugin/pkg/commands/workload_list.go:73 +0x197
github.com/vmware-tanzu/apps-cli-plugin/pkg/cli-runtime.ExecE.func1(0xc000ce0280, {0x54363e8, 0x0, 0x0})
	/Users/adhol/projects/apps-cli-plugin/pkg/cli-runtime/options.go:70 +0x12f
github.com/spf13/cobra.(*Command).execute(0xc000ce0280, {0x54363e8, 0x0, 0x0})
	/Users/adhol/projects/apps-cli-plugin/vendor/github.com/spf13/cobra/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc000ca2c80)
	/Users/adhol/projects/apps-cli-plugin/vendor/github.com/spf13/cobra/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/adhol/projects/apps-cli-plugin/vendor/github.com/spf13/cobra/command.go:902
github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/command/plugin.(*Plugin).Execute(...)
	/Users/adhol/projects/apps-cli-plugin/vendor/github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/command/plugin/plugin.go:54
main.main()
	/Users/adhol/projects/apps-cli-plugin/cmd/plugin/apps/main.go:127 +0xa91
Error: exit status 2

✖  exit status 2
```


### Expected behavior
```
Error: Unable to connect to k8s cluster at "https://127.0.0.1:50471/api?timeout=32s": dial tcp 127.0.0.1:50471: connect: connection refused
Error: exit status 1

✖  exit status 1
```

### Screenshots
![Screen Shot 2022-03-31 at 1 33 48 PM](https://user-images.githubusercontent.com/7171846/161115649-bb94a8fe-3d53-4768-a2e9-593948f291d5.png)
![Screen Shot 2022-03-31 at 1 34 14 PM](https://user-images.githubusercontent.com/7171846/161115735-e828c233-062e-4e03-8593-0c9eb360ac7a.png)


### Version (Apps plugin version, Version of K8s running on cluster)
Apps plugin v0.5.1

### Environment where the bug was observed (cloud, OS, etc)
macOS Monterey v12.3

--- a/pkg/cli-runtime/client.go
+++ b/pkg/cli-runtime/client.go
@@ -19,6 +19,7 @@ package cli
 import (
 	"context"
 	"fmt"
+	"os"
 
 	"github.com/go-logr/logr"
 	"k8s.io/apimachinery/pkg/api/meta"
@@ -29,6 +30,8 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/client-go/util/flowcontrol"
 	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/vmware-tanzu/apps-cli-plugin/pkg/cli-runtime/printer"
 )
 
 var (
@@ -179,7 +182,13 @@ func (c *client) lazyLoadRestConfigOrDie() *rest.Config {
 		kubeConfig := c.lazyLoadKubeConfig()
 		restConfig, err := kubeConfig.ClientConfig()
 		if err != nil {
-			panic(err)
+			if clientcmd.IsEmptyConfig(err) {
+				fmt.Printf("%s Unable to connect: no configuration has been found. If a kubeconfig is not set, it can be provided with the --kubeconfig flag or KUBECONFIG environment variable.\n", printer.Serrorf("Error:"))
+			} else {
+				fmt.Printf("%s %v \n", printer.Serrorf("Error:"), err)
+			}
+			c.logError(err)
+			os.Exit(2)
 		}
 		restConfig.RateLimiter = flowcontrol.NewTokenBucketRateLimiter(qps, burst)
 		c.restConfig = restConfig
@@ -200,7 +209,9 @@ func (c *client) lazyLoadClientOrDie() crclient.Client {
 		restConfig := c.lazyLoadRestConfigOrDie()
 		client, err := crclient.New(restConfig, crclient.Options{Scheme: c.scheme})
 		if err != nil {
-			panic(err)
+			fmt.Printf("%s Unable to connect: connection refused. Confirm kubeconfig details and try again.\n", printer.Serrorf("Error:"))
+			c.logError(err)
+			os.Exit(2)
 		}
 		c.client = client
 	}
@@ -212,7 +223,14 @@ func (c *client) lazyLoadDefaultNamespaceOrDie() string {
 		kubeConfig := c.lazyLoadKubeConfig()
 		namespace, _, err := kubeConfig.Namespace()
 		if err != nil {
-			panic(err)
+			if clientcmd.IsEmptyConfig(err) {
+				fmt.Printf("%s Unable to connect: no configuration has been found. If a kubeconfig is not set, it can be provided with the --kubeconfig flag or KUBECONFIG environment variable.\n", printer.Serrorf("Error:"))
+			} else {
+				fmt.Printf("%s %v \n", printer.Serrorf("Error:"), err)
+			}
+
+			c.logError(err)
+			os.Exit(2)
 		}
 		c.defaultNamespace = namespace
 	}

